ios/RNTypeValidation.m

```diff
@@ -0,0 +1,211 @@
+        return @"siteId is not valid, siteId should be a string";
+    }
+    
+    if (![self isAmountDictionaryValid: configuration[@"amount"]]) {
+        return @"amount is not valid, amount should be a dictionary of type JPAmount";
+    }
+    
+    if (![self isReferenceValid:configuration[@"reference"]]) {
+        return @"reference is not valid, reference should be a dictionary of type JPReference";
+    }
+    
+    if (![configuration[@"cardAddress"] isKindOfClass: [NSMutableDictionary class]] && configuration[@"cardAddress"] != nil) {
+        return @"cardAddress is not valid, cardAddress should be a dictionary of type JPAddress";
+    }
+    
+    if (![self isUIConfigurationsValid:configuration[@"uiConfiguration"]]){
+        return @"uiConfiguration is not valid, uiConfiguration should be a dictionary of type JPUIConfiguration";
+    }
+    
+    if (![configuration[@"primaryAccountDetails"] isKindOfClass: [NSMutableDictionary class]] && configuration[@"primaryAccountDetails"] != nil) {
+        return @"primaryAccountDetails is not valid, primaryAccountDetails should be a dictionary of type JPPrimaryAccountDetails";
+    }
+    
+    if (![configuration[@"paymentMethods"] isKindOfClass: [NSNumber class]] && configuration[@"paymentMethods"] != nil) {
+        return @"paymentMethods is not valid, paymentMethods should be a number";
+    }
+    
+    if (![configuration[@"supportedCardNetworks"] isKindOfClass: [NSNumber class]] && configuration[@"supportedCardNetworks"] != nil) {
+        return @"supportedCardNetworks is not valid, supportedCardNetworks should be a number";
+    }
+    
+    return nil;
+}
+
++ (nullable NSString *)isApplePayConfigurationValid: (NSMutableDictionary *) properties {
+    NSDictionary *configuration = properties[@"configuration"];
+    NSString *isConfigurationValid = [self isConfigurationValid:properties];
+    if (isConfigurationValid) {
+        return isConfigurationValid;
+    }
+    if (![configuration[@"applePayConfiguration"] isKindOfClass: [NSMutableDictionary class]]) {
+        return @"applePayConfiguration is not valid, make sure applePayConfiguration corespond to JPApplePayConfiguration model";
+    } else {
+        NSMutableDictionary *applePayConfiguration = configuration[@"applePayConfiguration"];
+        
+        if (![applePayConfiguration[@"merchantId"] isKindOfClass:[NSString class]]) {
+            return (@"applePayConfiguration -> merchantId is not valid, it should be a string");
+        }
+        
+        if(![applePayConfiguration[@"countryCode"] isKindOfClass:[NSString class]]) {
+            return(@"applePayConfiguration -> countryCode is not valid, it should be a string");
+        }
+        
+        if(![self isPaymentSummaryItemsValid:applePayConfiguration[@"paymentSummaryItems"]]) {
+            return (@"applePayConfiguration -> paymentSummaryItems is not valid, it should be an array of PaymentSummaryItems type");
```

Comment:
Not meaningful. Payment Summary Item is a complex object that needs additional validation of its properties